### PR TITLE
[feat] Populate the map with project markers

### PR DESCRIPTION
**Pull request by ethan-tam33**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- added markers to google map by creating addMarker.tsx dependent on approval status
- added Project type object

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
Screenshot 2024-10-13 at 4 33 51 PM

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

- review the app/page.tsx file
- review the api/maps/addMarkers.tsx file
- review the types/helper.d.ts file
- load the home screen to see that all the markers with approval render properly (ensure that you have the API key in .env.local)

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @itsliterallymonique

### CODE REVIEW

**Comment by itsliterallymonique** [comment]
beautiful work! approved

**Review by itsliterallymonique — APPROVED** [review]
beautiful work! approved

**Comment by itsliterallymonique** [comment]
@ethan-tam33 Make sure you don't overwrite Julee's schema changes when you squash and merge! Thanks!

File: api/maps/addMarker.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Marker } from '@react-google-maps/api';
+import { Project } from '../../types/helper';
+
+export default function addMarker(projects: Project[] | null) {
+  return projects?.map((project, index) => {
+    return project.approved ? (
+      <Marker
+        key={index}
```
Comment (jjstnlee) [inline api/maps/addMarker.tsx:8]: Not 100% sure why, but I think it's not recommended to use index as the key for map functions. We should probably use the project id since it uniquely identifies each project! [https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/](url)

File: types/helper.d.ts
Comment (jjstnlee) [inline types/helper.d.ts:1]: If you look at /api/supabase/queries/schema.ts, there's already a Project type defined so I think you can use that instead of creating it again here. It looks like it's missing zipcode but you or someone can probably add that :3

File: components/Map/map.tsx [hunk 2]
```diff
@@ -2,20 +2,22 @@
 
 import { useCallback, useState } from 'react';
 import { GoogleMap, useJsApiLoader } from '@react-google-maps/api';
+import addMarker from '../../api/maps/addMarker';
+import { Project } from '../../types/helper';
```
Comment (jjstnlee) [inline components/Map/map.tsx:6]: import project from /api/supabase/query/schema.ts instead (see other comment)

File: app/testing/page.tsx [hunk 3]
```diff
@@ -4,25 +4,8 @@ import { CSSProperties, useEffect, useState } from 'react';
...
-  town: string;
-  region: string;
-  state_senate_district: number;
-  assembly_district: number;
-  project_image: string | null;
-  additional_information: string | null;
-  key_development_milestones: object | null;
-}
+import Map from '../../components/Map/map';
+import { Project } from '../../types/helper';
```
Comment (jjstnlee) [inline app/testing/page.tsx:8]: import project from /api/supabase/queries/schema.ts instead (see other comment)

File: app/page.tsx [hunk 4]
```diff
@@ -1,8 +1,23 @@
-import { CSSProperties } from 'react';
+'use client';
+
+import { CSSProperties, useEffect, useState } from 'react';
 import Map from '@/components/Map/map';
 import ProjectModal from '@/components/ProjectModal';
+import queryProjects from '../api/supabase/queries/query';
+import { Project } from '../types/helper';
```
Comment (jjstnlee) [inline app/page.tsx:7]: import project from /api/supabase/queries/schema.ts instead (see other comment)

File: api/maps/addMarker.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Marker } from '@react-google-maps/api';
+import { Project } from '../../types/helper';
```
Comment (jjstnlee) [inline api/maps/addMarker.tsx:2]: import project from /api/supabase/queries/schema.ts instead (see other comment)

**Review by jjstnlee — CHANGES_REQUESTED** [review]
great job ethan!!! i wrote some comments so just take a look at those when you can and also if i am wrong abt something pls lmk i am not a genius :(

File: types/helper.d.ts
Comment (itsliterallymonique) [reply types/helper.d.ts:1]: @ethan-tam33 I agree. Please pull from the repo again and you should see there is already a schema.d.ts file. It is in /types

File: api/maps/addMarker.tsx (hunk 1 above)
Comment (itsliterallymonique) [reply api/maps/addMarker.tsx:8]: yes, please use **project_id** instead of **index**. This will also help with future sprints when we have to pull data for specific projects since we can use the unique project_id for this!

File: types/helper.d.ts
Comment (ethan-tam33) [reply types/helper.d.ts:1]: gotcha!

File: api/maps/addMarker.tsx (hunk 1 above)
Comment (ethan-tam33) [reply api/maps/addMarker.tsx]: fixed!

**Comment by ethan-tam33** [comment]
Thank you Monique and Justin! Just fixed all the issues mentioned above in the most recent push.

**Comment by itsliterallymonique** [comment]
> Thank you Monique and Justin! Just fixed all the issues mentioned above in the most recent push.

@ethan-tam33 thanks. you can squash + merge to main now

**Review by jjstnlee — APPROVED** [review]
awesome sauce
